src/librustdoc/doctest.rs

```diff
@@ -412,8 +428,13 @@ pub fn make_test(
             // Any errors in parsing should also appear when the doctest is compiled for real, so just
             // send all the errors that librustc_ast emits directly into a `Sink` instead of stderr.
             let sm = Lrc::new(SourceMap::new(FilePathMapping::empty()));
+            supports_color =
+                EmitterWriter::stderr(ColorConfig::Auto, None, false, false, Some(80), false)
+                    .supports_color();
```

Comment:
Whether we support color does not depend on the test so this isn't the correct place to check this.

It's also not the correct way to check it. First of all the output for tests goes to `stdout` not `stderr`. Also the output is handled by the `test` crate which doesn't use an `Emitter`.

Reply:
It doesn't really matter: `EmitterWriter` provides a method which allows to detect if the output stream support colors. So even if the `test` crate doesn't use an `Emitter`, it still relies on `termcolor` and we need to tell it if we want colors or not.

As for `stderr`/`stdout`, I didn't see a case yet where they were handled differently. But if you have one, I'm very curious about it (always interesting to discover new corner cases).